crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -72,22 +72,24 @@ def transition(inst, name, state)
     #
     if state == "discovered"
       @logger.debug("Provisioner transition: discovered state for #{name} for #{state}")
-      db = Proposal.where(barclamp: "provisioner", name: inst).first
 
       #
       # Add the first node as the provisioner server
       #
       if role.override_attributes["provisioner"]["elements"]["provisioner-server"].nil?
         @logger.debug("Provisioner transition: if we have no provisioner add one: #{name} for #{state}")
-        add_role_to_instance_and_node("provisioner", inst, name, db, role, "provisioner-server")
+        Proposal.where(barclamp: "provisioner", name: inst).first do |db|
+          add_role_to_instance_and_node("provisioner", inst, name, db, role, "provisioner-server")
+        end
 
-        # Reload the roles
-        db.reload
         role = RoleObject.find_role_by_name "provisioner-config-#{inst}"
       end
 
+      result = false
       @logger.debug("Provisioner transition: Make sure that base is on everything: #{name} for #{state}")
-      result = add_role_to_instance_and_node("provisioner", inst, name, db, role, "provisioner-base")
+      Proposal.where(barclamp: "provisioner", name: inst).first do |db|
+        result = add_role_to_instance_and_node("provisioner", inst, name, db, role, "provisioner-base")
```

Comment:
Metrics/LineLength: Line is too long. [103/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)